packages/zosfiles/src/methods/upload/Upload.ts

```diff
@@ -487,6 +488,12 @@ export class Upload {
             apiResponse.etag = uploadRequest.response.headers.etag;
         }
 
+        if (options.encoding != null) {
+            await Utilities.chtag(session, origUssname, Tag.TEXT, options.encoding);
+        } else if (options.binary) {
+            await Utilities.chtag(session, origUssname, Tag.BINARY);
+        }
+
         return {
             success: true,
             commandResponse: ZosFilesMessages.dataSetUploadedSuccessfully.message,
```

Comment:
also, not sure if we should make this change... but the apiResponse is empty 😋 
Should we at least specify which file the buffer was uploaded to? and maybe the contents of the buffer too? 😅 

<img width="491" alt="image" src="https://github.com/user-attachments/assets/f4c1883f-036e-421d-95e1-a2d37bdd3039">

Reply:
Curious about your thoughts on this, @jace-roell?